components/permissions/permission_lifetime_utils.cc

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/permissions/permission_lifetime_utils.h"
+
+#include <utility>
+
+#include "base/i18n/time_formatting.h"
+#include "base/stl_util.h"
+#include "components/grit/brave_components_strings.h"
+#include "ui/base/l10n/l10n_util.h"
+
+namespace permissions {
+
+PermissionLifetimeOption::PermissionLifetimeOption(
+    base::string16 label,
+    base::Optional<base::TimeDelta> lifetime)
+    : label(std::move(label)), lifetime(std::move(lifetime)) {}
+
+PermissionLifetimeOption::PermissionLifetimeOption(
+    const PermissionLifetimeOption&) = default;
+PermissionLifetimeOption& PermissionLifetimeOption::operator=(
+    const PermissionLifetimeOption&) = default;
+PermissionLifetimeOption::PermissionLifetimeOption(
+    PermissionLifetimeOption&&) noexcept = default;
+PermissionLifetimeOption& PermissionLifetimeOption::operator=(
+    PermissionLifetimeOption&&) noexcept = default;
+PermissionLifetimeOption::~PermissionLifetimeOption() = default;
+
+std::vector<PermissionLifetimeOption> CreatePermissionLifetimeOptions() {
```

Comment:
This function should have tests, but it will be definitely changed in the upcoming PR when we figure out the actual options and the text format we want to present to the user.